Assets/Mirror/Core/NetworkConnectionToClient.cs

```diff
@@ -50,9 +52,29 @@ public class NetworkConnectionToClient : NetworkConnection
         /// <summary>Round trip time (in seconds) that it takes a message to go server->client->server.</summary>
         public double rtt => _rtt.Value;
 
+        public NetworkConnectionToClient(int networkConnectionId, string address)
+            : base(networkConnectionId)
+        {
+            this.address = address;
+
+            // initialize EMA with 'emaDuration' seconds worth of history.
+            // 1 second holds 'sendRate' worth of values.
+            // multiplied by emaDuration gives n-seconds.
+            driftEma = new ExponentialMovingAverage(NetworkServer.sendRate * NetworkClient.snapshotSettings.driftEmaDuration);
+            deliveryTimeEma = new ExponentialMovingAverage(NetworkServer.sendRate * NetworkClient.snapshotSettings.deliveryTimeEmaDuration);
+
+            // buffer limit should be at least multiplier to have enough in there
+            snapshotBufferSizeLimit = Mathf.Max((int)NetworkClient.snapshotSettings.bufferTimeMultiplier, snapshotBufferSizeLimit);
+        }
+
+        // keep the old contstructor for a while in order to not break all projects.
+        // DEPRECATED 2024-05-16
+        [Obsolete("'new NetworkConnection(connectionId)' constructor was changed to 'new NetworkConnection(connectionId, address)'")]
         public NetworkConnectionToClient(int networkConnectionId)
             : base(networkConnectionId)
         {
+            this.address = Transport.active.ServerGetClientAddress(connectionId);
```

Comment:
Why can't we just do this internally instead of the breaking changes?

Reply:
we could. 
but threaded transports can't provide ServerGetClientAddress, so it would be easier to always pass it in OnConnected instead.